libdevcrypto/SecretStore.cpp

```diff
@@ -84,35 +86,37 @@ static js::mValue upgraded(std::string const& _s)
 	return js::mValue();
 }
 
-SecretStore::SecretStore(std::string const& _path): m_path(_path)
+SecretStore::SecretStore(string const& _path): m_path(_path)
 {
-	load();
+	if (!m_path.empty())
```

Comment:
Please remove this condition and let create_directories throw. I don't see how the condition is necessary; if it is, consider a ensureFileStore() method which calls create_directories and is called by constructor and save(). This check is inevitable so I can't see a reason to defer the call. If load() works here, then save() *should* work later. This class is useless if it can't ensure read/write to m_path; thus, a good idea to check/ensure in the constructor.